### PR TITLE
Deprecate "paritydb-experimental" CLI in favour or "paritydb"

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: client/cli/src/arg_enums.rs [hunk 1]
```diff
@@ -212,6 +214,8 @@ impl std::str::FromStr for Database {
 		if s.eq_ignore_ascii_case("rocksdb") {
 			Ok(Self::RocksDb)
 		} else if s.eq_ignore_ascii_case("paritydb-experimental") {
+			Ok(Self::ParityDbDeprecated)
```
Comment (arkpar) [inline client/cli/src/arg_enums.rs:217]: This function gets called multiple times, so can't just print a warning here.

File: client/cli/src/config.rs [hunk 2]
```diff
@@ -230,6 +230,13 @@ pub trait CliConfiguration<DCV: DefaultConfigurationValues = ()>: Sized {
 		Ok(match database {
 			Database::RocksDb => DatabaseSource::RocksDb { path: rocksdb_path, cache_size },
 			Database::ParityDb => DatabaseSource::ParityDb { path: paritydb_path },
+			Database::ParityDbDeprecated => {
+				eprintln!(
```
Comment (arkpar) [inline client/cli/src/config.rs:234]: Using `log` here is problematic because it is not initialised yet.

File: client/cli/src/arg_enums.rs [hunk 3]
```diff
@@ -201,8 +201,10 @@ pub enum Database {
 	/// ParityDb. <https://github.com/paritytech/parity-db/>
 	ParityDb,
 	/// Detect whether there is an existing database. Use it, if there is, if not, create new
-	/// instance of paritydb
+	/// instance of RocksDb
```
Comment (cheme) [inline client/cli/src/arg_enums.rs:204]: Isn't it parity db ?
(that is what I got when testing the polkadot parity-db pr, actually I did observe an artifact of rocksdb but it was not used)

File: client/cli/src/arg_enums.rs [hunk 4]
```diff
@@ -224,7 +228,7 @@ impl std::str::FromStr for Database {
 impl Database {
 	/// Returns all the variants of this enum to be shown in the cli.
 	pub fn variants() -> &'static [&'static str] {
-		&["rocksdb", "paritydb-experimental", "auto"]
+		&["rocksdb", "paritydb", "paritydb-experimental", "auto"]
```
Comment (cheme) [inline client/cli/src/arg_enums.rs:231]: I would just remove paritydb-experimental from the list (not sure if it is working).

File: client/cli/src/arg_enums.rs (hunk 3 above)
Comment (arkpar) [reply client/cli/src/arg_enums.rs]: You are right. I though "Auto" is now the default CLI setting and it defaults to paritydb. But apparently "RocksDb" is still  default.

File: client/cli/src/arg_enums.rs (hunk 4 above)
Comment (arkpar) [reply client/cli/src/arg_enums.rs:231]: It does not work. Clap produces an error if an unknown variant is used.

**Review by cheme — APPROVED** [review]
Looks good.

File: client/cli/src/arg_enums.rs (hunk 1 above)
Comment (bkchr) [reply client/cli/src/arg_enums.rs:217]: We could only log it once using `OnceCell` for example

**Comment by arkpar** [comment]
bot merge

**Comment by Mitchell-Grimes** [comment]
Does this PR imply that `paritydb` may be used for production environments? Or are there still risk factors associated with it that are considered experimental still and may not be ready for production use?

**Pull request merged**